Update index.md (#7347)
MystenLabs/sui: doc/src/build/programming-with-objects/index.md

- [Chapter 4: Object Wrapping](../../build/programming-with-objects/ch4-object-wrapping.md)
  - Wrapping objects in another object.
- [Chapter 5: Dynamic Fields](../../build/programming-with-objects/ch5-dynamic-fields.md)

>   - Extending objects with fields referencing other objects which can be dynamically added, accessed, and removed.

Reviewer: This is worded in a way that could mean either that the object or the field can be added, accessed, and removed dynamically depending on how I interpret it. Is this more clear?
```suggestion
  - Extending objects with fields that reference other objects, and that you can add, access, and remove dynamically.
```
